src/activities/Elsa.Activities.Reflection/Activities/ExecuteFunction.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Elsa.Attributes;
+using Elsa.Results;
+using Elsa.Services;
+using Elsa.Services.Models;
+
+namespace Elsa.Activities.Reflection.Activities
+{
+    /// <summary>
+    /// Execute a Method by reflection.
+    /// </summary>
+    [ActivityDefinition(
+        Category = "Reflection",
+        Description = "Execute a Method by reflection.",
+        RuntimeDescription = "a => !!a.state.variableName ? `Execute a Method by reflection and store the result into <strong>${ a.state.variableName }</strong>.` : 'Execute a Method by reflection.'",
+        Outcomes = new[] { OutcomeNames.Done }
+    )]
+    public class ExecuteMethod : Activity
+    {
+
+        public ExecuteMethod(IWorkflowExpressionEvaluator evaluator)
+        {
+
+        }
+
+
+        [ActivityProperty(Hint = "The variables to use as parameters, seperated by comma in order of method call.")]
+        public string InputVariableNames
+        {
+            get => GetState<string>(null, "InputVariableNames");
+            set => SetState(value, "InputVariableNames");
+        }
+
+        [ActivityProperty(Hint = "The name of the variable to store the returned value into.")]
+        public string OutputVariableName
+        {
+            get => GetState<string>(null, "OutputVariableName");
+            set => SetState(value, "OutputVariableName");
+        }
+
+        [ActivityProperty(Hint = "Assembly name (fullname or filename) to load")]
+        public string AssemblyName
+        {
+            get => GetState<string>(null, "AssemblyName");
+            set => SetState(value, "AssemblyName");
+        }
+
+        [ActivityProperty(Hint = "Class name to start or lookup")]
+        public string ClassName
+        {
+            get => GetState<string>(null, "ClassName");
+            set => SetState(value, "ClassName");
+        }
+
+        [ActivityProperty(Hint = "Class is a static class?")]
+        public bool IsStaticClass
+        {
+            get => GetState<bool>(null, "IsStaticClass");
+            set => SetState(value, "IsStaticClass");
+        }
+
+        [ActivityProperty(Hint = "Method name to execute")]
+        public string MethodName
+        {
+            get => GetState<string>(null, "MethodName");
+            set => SetState(value, "MethodName");
+        }
+
+        protected override async Task<ActivityExecutionResult> OnExecuteAsync(
+            WorkflowExecutionContext context,
+            CancellationToken cancellationToken)
+        {
+            await Task.Delay(1);
+            var inputValues = InputVariableNames.Split(',').Select(s => context.GetVariable(s)).ToArray();
+            return Execute(context, inputValues);
+        }
+
+        protected override ActivityExecutionResult OnResume(WorkflowExecutionContext context)
```

Comment:
I'm not sure I see why you need to implement `OnResume`. Does the activity return a `HaltResult` somewhere?

Reply:
Nope, dont think so... Just a n00bish copy from another module. Can be removed?